.github/workflows/ci.yml

```
@@ -995,6 +995,23 @@ jobs:
         run: cargo check-external-types --all-features
         working-directory: tokio
 
+  check-unexpected-lints-cfgs:
+    name: check unexpected lints and cfgs
+    needs: basics
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust ${{ env.rust_nightly }}
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.rust_nightly }}
+      - name: don't allow warnings
+        run: sed -i '/#!\[allow(unknown_lints, unexpected_cfgs)\]/d' */src/lib.rs */tests/*.rs
+      - name: check for unknown lints and cfgs
+        run: cargo check --all-features --tests
+        env:
+          RUSTFLAGS: -Dwarnings --check-cfg=cfg(loom) --check-cfg=cfg(tokio_unstable) --check-cfg=cfg(tokio_taskdump) --check-cfg=cfg(fuzzing) --check-cfg=cfg(mio_unsupported_force_poll_poll) --check-cfg=cfg(tokio_internal_mt_counters) --check-cfg=cfg(fs) --check-cfg=cfg(tokio_no_parking_lot) -Funexpected_cfgs -Funknown_lints
```

Can be simplified to just one `--check-cfg`:
```suggestion
          RUSTFLAGS: -Dwarnings --check-cfg=cfg(loom, tokio_unstable, tokio_taskdump, fuzzing, mio_unsupported_force_poll_poll, tokio_internal_mt_counters, fs, tokio_no_parking_lot) -Funexpected_cfgs -Funknown_lints
```

---

Or maybe it would be better to put those args in a `@path` file. Idk.

---

Done.